File: packages/shared-types/src/data-source.ts
export interface DataSource {
  id: string;
  workspaceId: string;
  deviceId: string;
  name: string;
  path: string;
  sourceType: 'local' | 'smb' | 'sync_folder' | 'agent_output';
  recursive: boolean;
  includeGlobs: string[];
  excludeGlobs: string[];
  scanMode: 'watch' | 'scheduled' | 'watch_and_scan';
  sensitivity: string;
  enabled: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateDataSourceRequest {
  name: string;
  path: string;
  sourceType: string;
  recursive?: boolean;
  includeGlobs?: string[];
  excludeGlobs?: string[];
  scanMode?: string;
  sensitivity?: string;
}

File: apps/hub-service/src/models/data_source.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DataSource {
    pub id: String,
    pub workspace_id: String,
    pub device_id: String,
    pub name: String,
    pub path: String,
    pub source_type: String,
    pub recursive: bool,
    pub include_globs: serde_json::Value,
    pub exclude_globs: serde_json::Value,
    pub scan_mode: String,
    pub sensitivity: String,
    pub enabled: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDataSourceRequest {
    pub name: String,
    pub path: String,
    pub source_type: String,
    pub recursive: Option<bool>,
    pub include_globs: Option<Vec<String>>,
    pub exclude_globs: Option<Vec<String>>,
    pub scan_mode: Option<String>,
    pub sensitivity: Option<String>,
}
